src/core/border_router/routing_manager.cpp

```diff
@@ -1941,13 +2006,21 @@ void RoutingManager::OmrPrefixManager::Evaluate(void)
 {
     OT_ASSERT(Get<RoutingManager>().IsRunning());
 
+    DetermineLocalPrefix();
     DetermineFavoredPrefix();
 
     // Decide if we need to add or remove our local OMR prefix.
 
-    if (mFavoredPrefix.IsEmpty())
+    if (mFavoredPrefix.IsEmpty() || !mFavoredPrefix.IsFavoredOver(mLocalPrefix))
```

Comment:
This is related to last comment (about removing `IsFavoredOver` version which just checks the preference).

- When there is no DHCP PD prefx, we do not want to replace an existing OMR prefix (in netdata) with our local **generated** prefix (even if it is smaller prefix and technically favored).
- But under DHCP6 we do want to check only if preference or gour local PD-based prefix is higher.